src/dmd/link.d

```diff
@@ -68,35 +68,46 @@ version (CRuntime_Microsoft)
   }
 }
 
-/****************************************
- * Write filename to cmdbuf, quoting if necessary.
- */
+/**
+Returns: true if the given filename needs a quote
+*/
+private bool optlinkNeedsQuote(const(char)[] filename)
+{
+    foreach (c; filename)
+    {
+        if (!isalnum(c) && c != '_' && c != '.' && c != '\\')
```

Comment:
Still conservative, but the existing version is even more conservative. Especially with respect to `\` string escaping changes with qoutes, so we should probably just stick to the exisiting test.